src/targets/__tests__/commitOnGitRepository.test.ts

```diff
@@ -0,0 +1,51 @@
+import { pushArchiveToGitRepository } from '../commitOnGitRepository';
+import childProcess from 'child_process';
+
+const execSyncSpy = jest.spyOn(childProcess, 'execSync');
+
+execSyncSpy.mockImplementationOnce(() => {
+  return Buffer.from('noop');
+});
+
+const mockClone = jest.fn();
+const mockCheckout = jest.fn();
+const mockRaw = jest.fn();
+const mockCommit = jest.fn();
+const mockAddTag = jest.fn();
+const mockPushTags = jest.fn();
+
+jest.mock('simple-git', () => () => ({
+  clone: mockClone,
+  checkout: mockCheckout,
+  raw: mockRaw,
+  commit: mockCommit,
+  addTag: mockAddTag,
+  pushTags: mockPushTags,
+}));
+
+test('Basic commit-on-git-repository functionality', async () => {
+  await pushArchiveToGitRepository({
+    archivePath: '/tmp/my-archive.tgz',
+    branch: 'main',
+    createTag: true,
+    repositoryUrl: 'git@github.com:getsentry/craft-test-repo.git',
+    stripComponents: 1,
+    version: '1.2.3',
+  });
+
+  expect(mockClone).toHaveBeenCalledWith(
+    'git@github.com:getsentry/craft-test-repo.git',
+    expect.any(String)
+  );
+  expect(mockCheckout).toHaveBeenCalledWith('main');
+  expect(mockRaw).toHaveBeenCalledWith('rm', '-r', '.');
+  expect(execSyncSpy).toHaveBeenCalledWith(
+    'tar -zxvf /tmp/my-archive.tgz --strip-components 1',
+    expect.objectContaining({ cwd: expect.any(String) })
+  );
+  expect(mockRaw).toHaveBeenCalledWith('add', '--all');
+  expect(mockCommit).toHaveBeenCalledWith('release: 1.2.3');
+  expect(mockAddTag).toHaveBeenCalledWith('1.2.3');
+  expect(mockRaw).toHaveBeenCalledWith('push', '--force');
+  expect(mockPushTags).toHaveBeenCalled();
+});
```

Comment:
this isn't a good validation -- it doesn't actually test anything (it's a "copy paste mock" sort of test where you've just copied the implementation to the test).  it's a fragile test that doesn't actually demonstrate the interesting behaviour of the functionality

please write a test which actually performs an extraction on a real tar and inspects the resulting git repository (you can skip the download / upload / push parts at least since those are the side-effects you can't test)

Reply:
I'd end up testing the simple-git library/tar which I don't see a case for. Use-Effort wise in my situation it doesn't make sense to add a testing setup for downloading release artifacts. I'm just gonna follow the convention in this repo which seems to be rather shallow testing.

This target is likely gonna be a one-off for a long time and the SDK team is happy to own it. If there are any issues we'll fix them and we will treat it as an incident if a wider scope is impacted.

Reply:
we're all trying to make craft better -- please follow up with an improved test